dashboard/modules/job/job_agent.py

```diff
@@ -0,0 +1,299 @@
+                        job_id)
+            return job_agent_pb2.InitializeJobEnvReply(
+                status=agent_manager_pb2.AGENT_RPC_STATUS_OK)
+
+        try:
+            job_description_data = json.loads(request.job_description)
+            job_info = JobInfo(
+                job_id=job_id,
+                temp_dir=self._dashboard_agent.temp_dir,
+                log_dir=self._dashboard_agent.log_dir,
+                **job_description_data)
+        except json.JSONDecodeError as ex:
+            error_message = str(ex)
+            error_message += f", job_payload:\n{request.job_description}"
+            logger.error("[%s] Initialize job environment failed, %s.", job_id,
+                         error_message)
+            return job_agent_pb2.InitializeJobEnvReply(
+                status=agent_manager_pb2.AGENT_RPC_STATUS_FAILED,
+                error_message=error_message)
+        except Exception as ex:
+            logger.exception(ex)
+            return job_agent_pb2.InitializeJobEnvReply(
+                status=agent_manager_pb2.AGENT_RPC_STATUS_FAILED,
+                error_message=traceback.format_exc())
+
+        async def _initialize_job_env():
+            os.makedirs(
+                job_consts.JOB_DIR.format(
+                    temp_dir=job_info.temp_dir, job_id=job_id),
+                exist_ok=True)
+            await DownloadPackage(job_info,
+                                  self._dashboard_agent.http_session).run()
+            logger.info("[%s] Starting driver.", job_id)
+            language = job_info.language
+            if language == job_consts.PYTHON:
+                driver = await StartPythonDriver(
+                    job_info, self._dashboard_agent.redis_address,
+                    self._dashboard_agent.redis_password).run()
+            else:
+                raise Exception(f"Unsupported language type: {language}")
+            job_info.driver = driver
+
+        initialize_task = create_task(_initialize_job_env())
+        self._job_table[job_id] = job_info
+
+        try:
+            await initialize_task
+        except asyncio.CancelledError:
+            logger.error("[%s] Initialize job environment has been cancelled.",
+                         job_id)
+            return job_agent_pb2.InitializeJobEnvReply(
+                status=agent_manager_pb2.AGENT_RPC_STATUS_FAILED,
+                error_message="InitializeJobEnv has been cancelled, "
+                "did you call CleanJobEnv?")
+        except Exception as ex:
+            logger.exception(ex)
+            return job_agent_pb2.InitializeJobEnvReply(
+                status=agent_manager_pb2.AGENT_RPC_STATUS_FAILED,
+                error_message=traceback.format_exc())
+
+        driver_pid = 0
+        if job_info.driver:
+            driver_pid = job_info.driver.pid
+
+        logger.info("[%s] Initialize job environment success.", job_id)
```

Comment:
`[job_id] Job environment initialized. And driver (pid=...) started.`

Reply:
Fixed.